### DISCUSSION TITLE
Why does `msrv.yml` specify `windows-2019`?

### DISCUSSION THREAD

**Discussion by EliahKagan** [GitoxideLabs/gitoxide#1514] [Q&A] [answered]
The MSRV workflow runs its Windows job on the oldest available Windows operating system with a GitHub-hosted runner, `windows-2019`. It occurred to me that this may be deliberate, to ensure things can be installed on an older system.

But it runs its Ubuntu job on the the rolling `ubuntu-latest` which is usually the most recent stable LTS release of Ubuntu (at the moment is second most recent such release).

https://github.com/Byron/gitoxide/blob/0b2829712be53846c830dbb175da2beaf00e7c76/.github/workflows/msrv.yml#L21-L23

What is the reason for this apparent disparity, and should one of them be changed?

**Comment by Byron** [comment, marked as answer]
This one I remember :)! Previously it was `windows-latest`, but then this switched from being 2019 to being 2022 or something, which broke the world. Thus it was pinned to the last known working version. This never happened with `ubuntu` so it's not pinned.
If there are no disadvantages, I think it could be pinned to the current stable version of Windows.

**Reply by EliahKagan** [reply]
Sounds good, I've opened #1515 for this.

My impression is that they are run on Windows in addition to Ubuntu to identify (and distinguish) issues that arise from the use of the `windows` crate, and that otherwise it would not be considered necessary to run them on multiple operating systems. Therefore, I have refrained from adding or proposing the addition of a job for macOS.
